src/Tasks/Microsoft.NET.Build.Tasks/ResolveReadyToRunCompilers.cs

```diff
@@ -141,7 +141,7 @@ private bool ValidateCrossgen2Support()
             }
 
             bool version5 = crossgen2PackVersion.Major < 6;
-            bool version6Preview1 = crossgen2PackVersion.Major == 6 && crossgen2PackVersion.Patch <= 21117;
+            bool version6Preview1 = crossgen2PackVersion.Major == 6 && crossgen2PackVersion.Release.StartsWith("preview.1.");
```

Comment:
How have you tested the `version6Preview1 == true` scenario?

Reply:
I'm testing it now with the "actual PR" for the fix, https://github.com/dotnet/sdk/pull/16029. According to Jeremy we should be regularly updating CG2 drop in the SDK repo but that's apparently not happening now, otherwise the SDK CI would be failing with the same bug as the dependency update. One way or another, if both this and the [NO-MERGE] PR's are green, that should suffice to ensure a smooth move forward for the SDK repo.

Reply:
The "actual PR" does not test `version6Preview1 == true` scenario, so I still have the same question: how have you tested this scenario?

Reply:
I don't understand, can you please clarify? What version of Crossgen2 is used today in the SDK repo according to your opinion? It cannot be Preview 2 because that has been fixed to use the CG1 perfmap name scheme (.ni.{MVID}.map) so the Ubuntu legs would be failing. It cannot be .NET 5 because in such case CG1 would be used for emitting the symbols.

Reply:
According to `eng/Version.Details.xml` file SDK is using `6.0.0-preview.2.21118.6` at the moment, not `preview1`.

Reply:
OK, apparently the 21118.6 version still uses the old naming as I verified by disassembling the ILCompiler.ReadyToRun.dll off the nuget package:

<pre>
    IL_0002:  call       string [System.Runtime]System.IO.Path::GetFileNameWithoutExtension(string)
    IL_0007:  ldstr      ".perf.map"
</pre>

The first nuget containing the new naming scheme is 21119.3:

<pre>
    IL_0000:  ldarga.s   perfMapMvid
    IL_0002:  call       instance bool valuetype [System.Runtime]System.Nullable`1<valuetype [System.Runtime]System.Guid>::get_HasValue()
    IL_0007:  brtrue.s   IL_0010

    IL_0009:  ldstr      "composite"
    IL_000e:  br.s       IL_0025

    IL_0010:  ldarga.s   perfMapMvid
    IL_0012:  call       instance !0 valuetype [System.Runtime]System.Nullable`1<valuetype [System.Runtime]System.Guid>::get_Value()
    IL_0017:  stloc.2
    IL_0018:  ldloca.s   V_2
    IL_001a:  constrained. [System.Runtime]System.Guid
    IL_0020:  callvirt   instance string [System.Runtime]System.Object::ToString()
    IL_0025:  stloc.0
    IL_0026:  ldarg.1
    IL_0027:  ldarg.2
    IL_0028:  call       string [System.Runtime]System.IO.Path::GetFileNameWithoutExtension(string)
    IL_002d:  ldstr      ".ni.{"
    IL_0032:  ldloc.0
    IL_0033:  ldstr      "}.map"
</pre>

I'll reflect this by using lexicographic comparison in the "actual PR".

Reply:
// using lexicographic comparison //
I still have the question "Why do we care?" What is a real scenario you are trying to support?

Reply:
Well, I guess the real scenario is merging in a green PR that remains green after the SDK dependency update. If you have different ideas on how to tackle that, I'm looking forward to hearing them.

Reply:
You could push the changes directly to the branch in #15978.

Reply:
OK, I have just done that.